### PR TITLE
feat: release date

### LINKED ISSUE #5822: Display age/release date for updates
**What would you like Renovate to be able to do?**

Give a clear indication to users about when an update was released (if we have that data)

**Describe the solution you'd like**

We might need to offer various approaches. Here are the possibilities I can think of:
- Raw release timestamp (accurate, but not user-friendly)
- Pretty timestamp (e.g. `16-Mar` if it's same year, or `12-Sep-2018` if it's older than a year)
- Relative age generated by Renovate at runtime (e.g. "2 days ago")
- Relative age generated by a hosted service using SVG (e.g. "2 days ago" but it's dynamically updated without the need for Renovate to run)

People will have different preferences, so we could satisfy this by exposing multiple values per-update to the PR body templates, e.g.

- `timestampRaw`: `20200320T140201Z`
- `timestampPrettyDay`: `16-Mar-2020`
- `timestampPrettyDayShort`: `16-Mar`
- `timestampAge`: `2 hours`

### PR DESCRIPTION
**Pull request by ViralRuparel**
Adds releaseTimeStamp to the PR updates table.

Closes #5822

### CODE REVIEW

**Comment by ViralRuparel** [comment]
Here is the result after adding timeStamp. @rarkins what so you think, should it be kept as show in the screenshot(in change column) or make a new column.(something like releaseTime).

image

**Comment by rarkins** [comment]
Shouldn’t be in any column by default yet and it needs to be formatted as discussed in the issue.

**Comment by rarkins** [comment]
Looks like you need to update your prettier version too btw

**Comment by ViralRuparel** [comment]
> Shouldn’t be in any column by default yet and it needs to be formatted as discussed in the issue.

Yeah I will format as we discussed to support all 4 formats, but just wanted to know the default placement of it in the PR(if not in any column of the table).

**Comment by rarkins** [comment]
No default placement yes but add it to allowed template fields

File: lib/config/definitions.ts [hunk 1]
```diff
@@ -1595,7 +1595,7 @@ const options: RenovateOptions[] = [
       Update: '{{{updateType}}}',
       'Current value': '{{{currentValue}}}',
       'New value': '{{{newValue}}}',
-      Change: '`{{{displayFrom}}}` -> `{{{displayTo}}}`',
+      Change: '`{{{displayFrom}}}` -> `{{{displayTo}}}` {{{releaseTimestamp}}}',
```
Comment (rarkins) [inline lib/config/definitions.ts:1598]:
```suggestion
      Change: '`{{{displayFrom}}}` -> `{{{displayTo}}}`',
```

revert

File: lib/manager/npm/index.ts [hunk 2]
```diff
@@ -14,6 +14,6 @@ export const defaultConfig = {
   versioning: npmVersioning.id,
   prBodyDefinitions: {
     Change:
-      "[{{#if displayFrom}}`{{{displayFrom}}}` -> {{else}}{{#if currentValue}}`{{{currentValue}}}` -> {{/if}}{{/if}}{{#if displayTo}}`{{{displayTo}}}`{{else}}`{{{newValue}}}`{{/if}}](https://renovatebot.com/diffs/npm/{{replace '/' '%2f' depName}}/{{{fromVersion}}}/{{{toVersion}}})",
+      "[{{#if displayFrom}}`{{{displayFrom}}}` -> {{else}}{{#if currentValue}}`{{{currentValue}}}` -> {{/if}}{{/if}}{{#if displayTo}}`{{{displayTo}}}`{{else}}`{{{newValue}}}`{{/if}}](https://renovatebot.com/diffs/npm/{{replace '/' '%2f' depName}}/{{{fromVersion}}}/{{{toVersion}}}) {{#if releaseTimestamp}}({{{releaseTimestamp}}}){{/if}}",
```
Comment (rarkins) [inline lib/manager/npm/index.ts:17]: revert this

File: lib/util/template/index.ts [hunk 3]
```diff
@@ -44,6 +44,7 @@ export const allowedFields = {
   depType: 'The dependency type (if extracted - manager-dependent)',
   displayFrom: 'The current value, formatted for display',
   displayTo: 'The to value, formatted for display',
+  releaseTimestamp: 'The release timeStamp of the newer version',
```
Comment (rarkins) [inline lib/util/template/index.ts:47]:
```suggestion
  releaseDate: 'The release date of the newer version',
```

File: lib/workers/pr/body/index.ts [hunk 4]
```diff
@@ -11,7 +11,7 @@ import { getControls } from './controls';
 import { BranchConfig } from '../../common';
 
 function massageUpdateMetadata(config: BranchConfig): void {
-  config.upgrades.forEach((upgrade) => {
+  config.upgrades.forEach(upgrade => {
```
Comment (rarkins) [inline lib/workers/pr/body/index.ts:14]: Update your `prettier` (e.g. run `yarn` to update node_modules) and then run `yarn prettier-fix` to revert this change.

File: lib/workers/pr/body/updates-table.ts [hunk 5]
```diff
@@ -13,6 +13,7 @@ function getTableDefinition(config: BranchConfig): TableDefinition[] {
     const value = config.prBodyDefinitions[header];
     res.push({ header, value });
   }
+
```
Comment (rarkins) [inline lib/workers/pr/body/updates-table.ts:16]: revert all changes in this file

**Review by rarkins — CHANGES_REQUESTED** [review]
- Do not make any changes to the default tables
- Instead, add a new field to upgrades called `releaseDate`, which is formatted like below:

If less than a year old, make it like `01-Jan`.
If more than a year old, make it like `01-Jan-2018`

File: lib/config/definitions.ts (hunk 1 above)
Comment (ViralRuparel) [reply lib/config/definitions.ts:1598]: sure

File: lib/manager/npm/index.ts (hunk 2 above)
Comment (ViralRuparel) [reply lib/manager/npm/index.ts:17]: okay

**Pull request closed without merging**